### DISCUSSION TITLE
Calendar Node (beta)

### DISCUSSION THREAD

**Discussion by zachowj** [zachowj/node-red-contrib-home-assistant-websocket#1295] [General]
https://zachowj.github.io/node-red-contrib-home-assistant-websocket/node/events-calendar.html

**Comment by rallep71** [comment]
First of all, thank you very much for the node! I have found the following: When I create an event in the calendar, it works in Node Red. However, if I change the times in the calendar for the same entry, it is sometimes triggered in Node Red and sometimes not.

**Comment by rallep71** [comment]
I have tested it again. When I create a new event in the calendar, it is triggered. However, if I only change the time for an event that has already been created, it is no longer triggered.

**Comment by yanikdude** [comment]
Hey All.

Using this new node and trying to get the Calendar entry details pushed to my phone. I can get the phone push notifications working but can't get the details to output. Any help would be greatly appreciated.

**Comment by lukejagger** [comment]
It doesn't seem to work some of the time.  I am searching for calendar event with 'Reserv' at the beginning, 10 hours prior to start, and I've seen it work once, but now it doesn't seem to trigger.  Will it look past other events as I have an 'Airbnb (Not available)' event before the 'Reservation' event.

**Comment by gbrogna** [comment]
I have been waiting for this node.  Like others have said it is intermittent and unreliable.  Looking forward to updates.  Thank you for all your efforts.

**Comment by lukejagger** [comment]
Any progress on fixing this, it is a great addition!

**Comment by morlie** [comment]
I couldn't find any detail in the current documentation. When I create an event in the google calendar it's not shown in Node-Red immediately. How long does it take till Node-Red sees the event? When I restart Node-Red it's showing the event immedtiately after it's online again.

The events are shown in Google and Homeassistant immediately but not queued in NR. Is there an option to trigger and re-load newly created events?

**Reply by Ewoodster** [reply]
Same problem here. Only after deploying, the node will update and queue the calendar event.
In general, the node is awesome, but it has been in Beta for a year now without any updates. I wonder if it will ever be released.

**Comment by nightfever77** [comment]
Negative offset does not seem to work as intended.
E.g. offset is set to -15 minutes, but payload will always be injected 15 minutes after event start.

Edit: Just saw in the issues that the offset is inverted. Can live with that, just need to know.

**Comment by Teme-V** [comment]
With recurring calendar event, i.e. daily but with 30d interval - node trigger previous event information on each day - next event in calendar is on 20250402T221500
```
recurrence_id: "20250303T221500"
rrule: "FREQ=DAILY;INTERVAL=30"
```

**Comment by RoberDNuno** [comment]
I need help with this. I can't reed events

**Reply by morlie** [reply]
output for input. You need to be more precisely. Whats your problem? Whats your expectation? Provide your configuration and what information or help you require.

**Reply by RoberDNuno** [reply]
Hello, I need to create an automation where the trigger is a DAILY event. If it exists, it sets a global variable with one value, and if it doesn't, it sets a different value.
When I implement the calendar node, it doesn't detect the event in the calendar.

![image](https://github.com/user-attachments/assets/358cc425-9c1f-4bcc-bfc1-f5467448fde6)

**Reply by RoberDNuno** [reply]
The event happens daily at 7:30 a.m., so I want to know if the event is going to happen by 5:30 a.m.
With small offsets, it works, but when I set it to 2 hours, it doesn't.

**Reply by yanikdude** [reply]
So you are tying to see if there is an event scheduled for the following day, which accrues at 07:30, to trigger another event at 05:30 like preheat the car?

Or I maybe worth specifying the full use case.

**Reply by RoberDNuno** [reply]
Let me explain this better:
At 5:30 a.m., I need to know if there is a calendar event that will occur at 7:30 a.m. (2-hour offset).
With the node I showed you earlier, it sometimes works with small offsets, but not with longer offsets.
When the node discovers if the event exists, it modifies a variable to perform different automations (lights, blinds, coffee, etc.)

**Reply by morlie** [reply]
It's a well known bug since a while
https://github.com/zachowj/node-red-contrib-home-assistant-websocket/issues/1302

**Reply by RoberDNuno** [reply]
Is a short-term solution expected? Are you using any alternatives?

**Reply by morlie** [reply]
> Is a short-term solution expected? Are you using any alternatives?
But was reported more than 1 year back. It's unlikely.

You could read the google calendar in Home Assistant and fire an event. Node-Red reacts on the event and does whatever needs to be done. Or you create the calender entry starting 5:30 ... 

`
description: ""
mode: single
triggers:
  - trigger: calendar
    entity_id: calendar.homeassistant
    event: start
    offset: "--2:0:0"
conditions: []
actions:
  - event: xxxx
    event_data: {}
`

**Comment by fredrikbuchanan** [comment]
no biggie but slightly annoying.
There is a mismatch between the documentation and the actual web config , the documentation references "conditions" but not "filters", in the web config  there is a "filtering" choice but nothing about "conditions".

**Comment by phlo2009** [comment]
May I suggest to have 2 outputs for the node with for start and the other for stop to identify the state transitions. Condition filters are not necessary as this can be handled more efficiently by using an external Switch node.

**Comment by etymotic** [comment]
Hmmm, I'm trying to wrap my head around this node.

I was thinking it would output either "on" or "off" based on whether or not you're currently inside a calendar event, but it outputs text based on the event so you can't use a Switch node to made decisions based on "on" or "off".

It also only outputs once -- either at the start or the end.

So, in typing this out, if I want to trigger something at the beginning of an event and also at the end of the event, I need *two* calendar nodes.

Easy enough -- I just had to wrap my head around the logic.

**Comment by garrathbond** [comment]
how does the filter work please

**Comment by phlo2009** [comment]
Ref version 0.77.2

I tested the start and end modes. Works great as expected.

Philip

**Comment by joezappie** [comment]
I have a local calendar for our trash schedule. I had existing events when installing node red, and tried recreating them, but no matter what Im getting "No upcoming events queued at: Sept 10:55". Any info on why this is?

<img width="276" height="76" alt="image" src="https://github.com/user-attachments/assets/6668dcea-5a3c-4c94-9ab5-c8a8bc210b92" />

I have the node red hacs custom integration installed and use it for HA buttons. 

<img width="622" height="393" alt="image" src="https://github.com/user-attachments/assets/037b0803-f5cd-4a41-ad08-a42084d6ccf0" />

<img width="519" height="699" alt="image" src="https://github.com/user-attachments/assets/cb4dd27b-7ae6-467b-9736-4bc5a4a0124e" />

**Reply by carlson993** [reply]
I also have the same issue

**Reply by etymotic** [reply]
This might be normal behavior. Based on my testing the node only shows events coming up within a pretty short time period (48 hours, maybe?). Newly added events won't show up immediately, I think the node polls the calendar every 15 minutes (or upon restarting Node-red.)

So if an event is scheduled more than 2 days out, it won't show up on Node-red. Make a test event within the next 48 hours and wait 15 minutes or so for the node to pick it up. It should show at that point.

Other things to note: Do you have two different calendars? There's a blue calendar for recycling and a yellow for trash. The node you show is referencing the "trash" calendar whose next event was 4 days out.

**Comment by thijssie909** [comment]
@zachowj , Somehow the node is outputting an event at the start and end, although start is selected. More people experiencing this?
